__init__.py

```
@@ -304,6 +331,11 @@ def process_tree(filecontent, file_tree, with_progress, with_header=False):
 
     if with_header:
         header = dict(map(make_header_ent, header.children[0].children))
+        for field in HEADER_FIELDS.keys():
+            observed = header.get(field.upper(), [])
+            expected = HEADER_FIELDS.get(field)._fields
+            if len(header.get(field.upper(), [])) != len(expected):
+                raise HeaderFieldError(field.upper(), len(observed), len(expected))
```

Do we want to really raise an exception and terminate or wait for potentially additional errors?